core/src/Cuda/Kokkos_Cuda_Instance.cpp

```diff
@@ -202,7 +202,9 @@ void cuda_internal_error_abort(cudaError e, const char *name, const char *file,
   if (file) {
     out << " " << file << ":" << line;
   }
-  abort(out.str().c_str());
+  // Call Kokkos::Impl::host_abort instead of Kokkos::abort to avoid a warning
+  // about Kokkos::abort returning in some cases.
+  host_abort(out.str().c_str());
```

Comment:
I am not sure about that one.  I don't like spreading use of internal details if we can avoid.  This would be the first use of `Impl::host_abort` in the code base besides the definition of `Kokkos::abort`.
What other option do we have?

Reply:
>  What other option do we have?

We could just keep this one. It's only compiled once anyway.

Reply:
Just prefix with FIXME and I will approve

Reply:
f6634bcd39e9bc04966ecb5a28ecbf4b22310086